src/main/java/duke/command/CommandType.java

```diff
@@ -0,0 +1,35 @@
+package duke.command;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum CommandType {
+    TODO("todo"),
+    DEADLINE("deadline"),
+    EVENT("event"),
+    BYE("bye"),
+    LIST("list"),
+    DONE("done"),
+    DELETE("delete");
+
+    private final String label;
+    private static final Map<String, CommandType> BY_LABEL = new HashMap<>();
+
+    static {
+        for (CommandType command : values()) {
+            BY_LABEL.put(command.label, command);
+        }
+    }
```

Comment:
This is the first time i'm seeing this syntax. How does this work?

Reply:
It is called a static block. Whenever a class is loaded into memory Java will execute the block only once.